tasks/bake.js

```diff
@@ -483,18 +500,26 @@ module.exports = function( grunt ) {
 				if ( oldValue === undefined ) values[ forEachName ] = oldValue;
 				else delete values[ forEachName ];
 
-				return fragment;
+				content = fragment;
 
 			} else if( !forEachName ) {
 
 				processExtraBake( extraBake, filePath, destFile, values );
 
-				return linebreak + parse( includeContent, includePath, destFile, values );
+				content = linebreak + parse( includeContent, includePath, destFile, values );
 
 			} else {
 
+				content = "";
+			}
+
+			if( assign !== null ) {
+				parentValues[ assign ] = mout.string.ltrim( content );
```

Comment:
Honestly, I have not thought intensively about fixing the indentation for assignments, the ltrm was to remove the prepended line break to prevent ending up with an additional empty line. To compute the right indent we need the indentation of the placeholder (`{{code}}`), don't we? And if yes, I have currently no idea how to get that (without bigger refactoring). Also it could be that one places the included code within a tag. I have not tested it, but assign should also work this way:

```HTML

<!--(bake-start _assign="class" _if="foo")-->active<!--(bake-end)-->
<a href="..." class="{{class}}">Click</a>
```